src/components/breadcrumb/OpenCommentsButton.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { CommentActions, IComment } from 'tdp_comments';
+
+export interface IOpenCommentsButtonProps {
+  /**
+   * idType of the underlying workbench data.
+   */
+  idType: string;
+
+  /**
+   * Workbench selection.
+   */
+  selection: string[];
+  /**
+   * CommentPanel visibility, used to display the correct title on hover.
+   */
+  commentPanelVisible: boolean;
+
+  /**
+   * Show/hide CommentPanel.
+   */
+  onCommentPanelVisibilityChanged: (open: boolean) => void;
+}
+
+export function OpenCommentsButton({ idType, selection, commentPanelVisible, onCommentPanelVisibilityChanged }: IOpenCommentsButtonProps) {
+  const [commentCount, setCommentCount] = React.useState<number>(0);
+
+  React.useEffect(() => {
+    const listener = (_, comments: IComment[]) => {
+      // TODO: should we count the replies? The original button from tdp_comments shows the count of unread comments and + replies
+      // comments.map((c) => c.replies.filter((r) => !r.read).length);
+      const count = comments.filter((c) => c.entities.some((e) => e.id_type === idType)).length;
+      setCommentCount(count);
+    };
+    CommentActions.onChangedComments(listener);
+    const visibilityListener = (_, visible: boolean) => onCommentPanelVisibilityChanged(visible);
+    CommentActions.onCommentPanelVisibiltyChanged(visibilityListener);
+
+    return () => {
+      CommentActions.offChangedComments(listener);
+      CommentActions.offCommentPanelVisibiltyChanged(visibilityListener);
+    };
+  }, [idType, onCommentPanelVisibilityChanged]);
+
+  React.useEffect(() => {
+    if (selection.length) {
+      return;
+    }
+    onCommentPanelVisibilityChanged(false);
+  }, [onCommentPanelVisibilityChanged, selection]);
+
+  const title = commentPanelVisible
+    ? 'Hide comments panel'
+    : commentCount
+    ? `${commentCount} new comments are available for your selection`
+    : 'Show comments panel';
```

Comment:
Extract language strings using i18n.

Reply:
This text has to also be adapted according to what we decide the count to be (all comments or new comments) in the above point

Reply:
Yes, I see the if-conditions.

```suggestion
    ? I18nextManager.getInstance().i18n.t('ordino:comments.hideCommentsPanel')
    : commentCount
    ? I18nextManager.getInstance().i18n.t('ordino:comments.newCommentsForSelection', {commentCount})`
    : I18nextManager.getInstance().i18n.t('ordino:comments.showCommentsPanel');
```

with the following language file:

```js
{
   "newCommentsForSelection": "{{commentCount}} new comments are available for your selection",
}
```